velox/expression/CastExpr.cpp

```diff
@@ -875,6 +884,78 @@ void CastExpr::applyPeeled(
             fromType, toType, rows, context, input, result);
         break;
     }
+  } else if (
+      isIntegralType(fromType) && isBasicNumericType(toType) &&
+      ((fromType->cppSizeInBytes() < toType->cppSizeInBytes()) ||
+       (fromType == INTEGER() && toType == REAL()) ||
+       (fromType == BIGINT() && toType == REAL()) ||
+       (fromType == BIGINT() && toType == DOUBLE()) ||
+       (fromType == HUGEINT() && toType == REAL()) ||
+       (fromType == HUGEINT() && toType == DOUBLE()))) {
```

Comment:
real to double also should be ok?

Reply:
Nice catch, thanks. I added its support as well as tests and benchmarks.
I opened https://github.com/facebookincubator/velox/issues/15506 for the discussion of avoiding "try catch" in cast.